### PR TITLE
[macOS] Add ability to open multiple editor instances and global/dock menu access

### PR DESCRIPTION
**Pull request by bruvzg**
Adds methods to access macOS dock and global menu.

Adds "New Windows" dock menu item (editor and project manager).
Adds favourite/recent project list to project manager dock menu.
Adds opened scene list to editor dock menu.

Double-clicking ".godot" files in Finder now open new instance as well.

screenshot 2019-02-07 at 00 05 04

### CODE REVIEW

**Comment by bruvzg** [comment]
Nice, clang-format is [suggesting change](https://travis-ci.org/godotengine/godot/jobs/489775321#L548) that's causing build to [fail on macOS](https://travis-ci.org/godotengine/godot/jobs/489770886#L173).

**Comment by akien-mga** [comment]
> Nice, clang-format is [suggesting change](https://travis-ci.org/godotengine/godot/jobs/489775321#L548) that's causing build to [fail on macOS](https://travis-ci.org/godotengine/godot/jobs/489770886#L173).

Hm, that's bad. The space in `> >` is needed in C++03, but we won't need it anymore when we move to C++11. The `.clang-format` file defines `Standard: Cpp03`, so it handles it properly in C++ files, but it seems like the ObjC++ files follow a different standard.

**Comment by akien-mga** [comment]
4912f57 should fix the `clang-format` issue.

**Comment by samgreen** [comment]
I don't have much to add other than a 👍 on this. Would be a big quality of life improvement for those of us building games on OS X.

**Comment by akien-mga** [comment]
The macOS-specific changes seem good to me.

CC @reduz and @hpvb to review the added OS APIs though, I wonder if it's fine to add those if they're only used on one platform.

**Comment by akien-mga** [comment]
> I wonder if it's fine to add those if they're only used on one platform.

This is still my main concern with this PR. Would it be possible to implement this feature ad hoc in the macOS platform, exposing the relevant configuration in the platform's export settings? I see that the global menu action should be defined from script, so I guess this wouldn't play well with export options :/

File: platform/osx/os_osx.mm [hunk 1]
```diff
@@ -61,6 +61,10 @@
 #include <sys/types.h>
 #include <unistd.h>
 
+#ifdef TOOLS_ENABLED
+#include "editor_settings.h"
```
Comment (akien-mga) [inline platform/osx/os_osx.mm]: This doesn't appear to be used IINM.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**